tests/test_app/test_data_store.py

```diff
@@ -305,6 +305,19 @@ def test_write_class_source_create_delete(self):
                 path, suffix=".dummySuffix", if_exists=OVERWRITE, create=True
             )
             self.assertEqual(len(dstore), 0)
+            # tests the case when the directory has log files
+            with open(
+                os.path.join(path, "test_write_class_source_create_delete.log"), "w"
+            ):
+                pass
+            with self.assertRaises(RuntimeError):
+                dstore = self.WriteClass(
+                    path, suffix=".json", if_exists=OVERWRITE, create=True
+                )
+            with self.assertRaises(RuntimeError):
+                dstore = self.WriteClass(
+                    path, suffix=".log", if_exists=OVERWRITE, create=True
```

Comment:
don't define the suffix as `.log`. The pattern is the data store can contain both the allowed suffix and members with name ending in `.log`, never only `.log` files.